### PR TITLE
added multilanguage version

### PR DESCRIPTION
**Pull request by mmarganski**

### CODE REVIEW

File: src/components/Main.tsx [hunk 1]
```diff
@@ -21,50 +20,29 @@ const TabComponents = {
...
-                        {tabComponent}
-                    </Route>
-                ))}
-                <Redirect to={Object.keys(TabComponents).find(Boolean) || ''}/>
-            </Switch>
-        </Router>
-    </Provider>
-)
+export const Main: React.FunctionComponent = () => {
+    const Translation: {[key: string]: {[key: string]: string}} = useTranslation()
```
Comment (gtarnopolski) [inline src/components/Main.tsx]: what is this my friend :D

File: src/components/RecipesView.tsx [hunk 2]
```diff
@@ -2,28 +2,32 @@ import React from 'react'
...
-)
+export const RecipesView: React.FunctionComponent<RecipesViewProps> = ({ recipes }) => {
+    const Translation = useTranslation()
+
+    return (
+        <ColumnWrapper>
+            <Header text={Translation.tabs.recipes}/>
+            <RowWrapper>
+                {recipes
+                    .map((name, index) => (
```
Comment (gtarnopolski) [inline src/components/RecipesView.tsx]:
```
  {recipes.map((name, index) => (
          <Item
              key={`${index}-${name}`}
              text={name}
              isWrapped={false}
          />
    ))}
```

File: src/components/StoreWrapper.tsx [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+import { Provider } from 'outstated'
+import { Main } from './Main'
+import { useProductsStore, useRecipesStore, useLanguageStore } from '../stores'
```
Comment (gtarnopolski) [inline src/components/StoreWrapper.tsx]: you can create stores array in stores folder, imagine if you will have 10 stores, import will look ugly

File: src/components/TabsBar.tsx [hunk 4]
```diff
@@ -1,30 +1,43 @@
...
-                key={`${index}-${tabName}`}
-                onSelect={() => onSelect(tabName)}
+        {tabs.map(tab => (
+            <StyledLink
+                to={`/${tab}`}
+                key={tab}
             >
-                <Header text={`${tabName}`}/>
-            </ClickableWrapper>
+                {translation?.[tab]}
```
Comment (gtarnopolski) [inline src/components/TabsBar.tsx]: what if there is now translation that suits tab

File: src/hooks/useTranslation.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { useStore } from 'outstated'
+import { useLanguageStore } from '../stores'
+
+export const useTranslation = () => {
```
Comment (gtarnopolski) [inline src/hooks/useTranslation.ts]: proxy

File: src/languages/LanguageNames.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+export enum LanguageNames {
+    eng = 'English',
+    pl = 'Polski',
```
Comment (gtarnopolski) [inline src/languages/LanguageNames.ts]: redundant semicolon

File: src/stores/useLanguageStore.ts [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { useEffect, useState } from 'react'
+import { LanguageDictionaries, LanguageNames } from '../languages'
+
+export const useLanguageStore = () => {
+
```
Comment (gtarnopolski) [inline src/stores/useLanguageStore.ts:5]: .

File: src/stores/useLanguageStore.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+    const [currentLanguage, setLanguage] = useState<LanguageNames>(
+        window.localStorage.getItem('language')
+            ? JSON.parse(window.localStorage.getItem('language') as string)
```
Comment (gtarnopolski) [inline src/stores/useLanguageStore.ts]: as LanguageNames

**Review by gtarnopolski — CHANGES_REQUESTED** [review]

File: src/components/ApiRecipesView.tsx [hunk 9]
```diff
@@ -17,6 +17,7 @@ export const ApiRecipesView: React.FunctionComponent<ApiRecipesViewProps> = ({
 }) => {
     const { fetchData, isLoading, hasError } = useFetch<Array<ApiResponseRecipe>>()
     const [recipes, setRecipes] = useState<Record<string, number>>({})
+    const Translation = useTranslation()
```
Comment (jpudysz) [inline src/components/ApiRecipesView.tsx]: lowercased `Translation`

File: src/components/ApiRecipesView.tsx [hunk 10]
```diff
@@ -38,19 +39,19 @@ export const ApiRecipesView: React.FunctionComponent<ApiRecipesViewProps> = ({
     const renderRecipes = () => {
         if (url === '') {
             return(
-                <Message text="select product to find matching recipes"/>
+                <Message text={Translation.messages.callToSelect}/>
```
Comment (jpudysz) [inline src/components/ApiRecipesView.tsx]: did you get auto-suggestions?

**Review by gtarnopolski — APPROVED** [review]
👍

File: src/components/Main.tsx [hunk 11]
```diff
@@ -21,50 +20,29 @@ const TabComponents = {
...
-                        {tabComponent}
-                    </Route>
-                ))}
-                <Redirect to={Object.keys(TabComponents).find(Boolean) || ''}/>
-            </Switch>
-        </Router>
-    </Provider>
-)
+export const Main: React.FunctionComponent = () => {
+    const Translation = useTranslation()
```
Comment (jpudysz) [inline src/components/Main.tsx]: aa

File: src/components/RecipesView.tsx [hunk 12]
```diff
@@ -2,28 +2,30 @@ import React from 'react'
...
 }
 
-export const RecipesView: React.FunctionComponent<RecipesViewProps> = ({ recipes }) => (
-    <ColumnWrapper>
-        <Header text={Tab.Recipes}/>
-        <RowWrapper>
-            {recipes
-                .map((name, index) => (
+export const RecipesView: React.FunctionComponent<RecipesViewProps> = ({ recipes }) => {
+    const Translation = useTranslation()
```
Comment (jpudysz) [inline src/components/RecipesView.tsx]: aa

File: src/components/TabsBar.tsx [hunk 13]
```diff
@@ -1,30 +1,43 @@
...
 import styled from 'styled-components'
-import { Header } from './Header'
-import { ClickableWrapper } from './ClickableWrapper'
-import { Tab } from '../types'
+import { Link } from 'react-router-dom'
 
 type TabsBarProps = {
-    onSelect(tab: Tab): void
+    tabs: Array<string>,
+    translation: {[key: string]: string}
```
Comment (jpudysz) [inline src/components/TabsBar.tsx]: `Record`

File: src/stores/useLanguageStore.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react'
+import { LanguageDictionaries, LanguageNames } from '../languages'
+
+export const useLanguageStore = () => {
+    const [currentLanguage, setLanguage] = useState<LanguageNames>(
+        window.localStorage.getItem('language')
```
Comment (jpudysz) [inline src/stores/useLanguageStore.ts]: no-magic-strings

**Review by jpudysz — CHANGES_REQUESTED** [review]

**Review by jpudysz — APPROVED** [review]

**Pull request merged**